can/logconvert.py

```diff
@@ -0,0 +1,69 @@
+"""
+Convert a log file from one format to another.
+"""
+
+import sys
+import argparse
+import errno
+
+from can import LogReader, Logger, SizedRotatingLogger
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        "python -m can.logconvert",
+        description="Convert a log file from one format to another.",
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output",
+        dest="output",
+        help="""Output filename, type dependent on suffix see can.LogReader.""",
+        default=None,
+        required=True,
+    )
+
+    parser.add_argument(
+        "-s",
+        "--file_size",
+        dest="file_size",
+        type=int,
+        help="""Maximum file size in bytes. Rotate log file when size threshold is reached.""",
+        default=None,
+    )
+
+    parser.add_argument(
+        "infile",
+        metavar="input-file",
+        type=str,
+        help="Log file to convert from. For supported types see can.LogReader.",
+    )
+
+    # print help message when no arguments were given
+    if len(sys.argv) < 2:
+        parser.print_help(sys.stderr)
+        raise SystemExit(errno.EINVAL)
+
+    results = parser.parse_args()
+
+    reader = LogReader(results.infile)
```

Comment:
We should probably close the file descriptors afterwards. `with LogReader(results.infile) as reader:` and similarly for the logger should be enough.

Reply:
Yes, will fix this.

Reply:
Seems like this is the only thing left to be done before merging.

Reply:
I solved this with a .stop() in the finally clause to avoid using contextmanagers or so due to the SizedRotatingLogger vs Logger instances. Will do the same thing as using `with`. Is that good enough?

Reply:
I commented below. I think you can just apply it as is and we can merge it. Functionally, they should do pretty much the same.